internal/keystore/operations_manager.go

```diff
@@ -0,0 +1,214 @@
+	}
+	if req.Provider == types.ProviderUnknown {
+		return types.OperationResult{}, ErrProviderRequired
+	}
+	if strings.TrimSpace(string(req.Name)) == "" {
+		return types.OperationResult{}, ErrOperationNameRequired
+	}
+
+	key := operationKey{
+		Provider: req.Provider,
+		Name:     req.Name,
+	}
+
+	m.mu.RLock()
+	descriptor, ok := m.descriptors[key]
+	m.mu.RUnlock()
+
+	if !ok {
+		return types.OperationResult{}, ErrOperationNotRegistered
+	}
+
+	payload, err := m.resolveCredential(ctx, req)
+	if err != nil {
+		return types.OperationResult{}, err
+	}
+
+	client, err := m.resolveClient(ctx, req, descriptor)
+	if err != nil {
+		return types.OperationResult{}, err
+	}
+
+	input := types.OperationInput{
+		OrgID:      req.OrgID,
+		Provider:   req.Provider,
+		Credential: payload,
+		Client:     client,
+		Config:     cloneConfigMap(req.Config),
+	}
+
+	result, runErr := descriptor.Run(ctx, input)
+	if result.Status == "" {
+		result.Status = types.OperationStatusUnknown
+	}
+
+	if runErr != nil {
+		return result, runErr
+	}
+
+	return result, nil
+}
+
+// resolveCredential retrieves or refreshes the credential based on the request flags
+func (m *OperationManager) resolveCredential(ctx context.Context, req types.OperationRequest) (types.CredentialPayload, error) {
+	if req.Force {
+		return m.source.Mint(ctx, req.OrgID, req.Provider)
+	}
+
+	return m.source.Get(ctx, req.OrgID, req.Provider)
+}
+
+// resolveClient retrieves a client instance if the operation requires one
+func (m *OperationManager) resolveClient(ctx context.Context, req types.OperationRequest, descriptor types.OperationDescriptor) (any, error) {
+	if descriptor.Client == "" {
+		return nil, nil
+	}
+
+	if m.clients == nil {
+		return nil, ErrOperationClientManagerRequired
+	}
+
+	opts := []ClientRequestOption[map[string]any]{}
+	if len(req.Config) > 0 {
+		opts = append(opts, WithClientConfig(maps.Clone(req.Config)))
+	}
+	if req.ClientForce {
+		opts = append(opts, WithClientForceRefresh[map[string]any]())
+	}
+
+	return m.clients.Get(ctx, req.OrgID, req.Provider, descriptor.Client, opts...)
+}
+
+// Descriptors returns a copy of all registered operations keyed by provider
+func (m *OperationManager) Descriptors() map[types.ProviderType][]types.OperationDescriptor {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+
+	grouped := map[types.ProviderType][]types.OperationDescriptor{}
+	for key, descriptor := range m.descriptors {
+		grouped[key.Provider] = append(grouped[key.Provider], descriptor)
+	}
+
+	for provider, descriptors := range grouped {
+		copied := make([]types.OperationDescriptor, len(descriptors))
+		copy(copied, descriptors)
+		grouped[provider] = copied
+	}
+
+	return grouped
+}
+
+// operationKey uniquely identifies an operation by provider and name
+type operationKey struct {
+	// Provider identifies which provider publishes the operation
+	Provider types.ProviderType
+	// Name identifies the specific operation within the provider
+	Name types.OperationName
```

Comment:
shouldn't need to be exported since the struct isn't

```suggestion
	// Provider identifies which provider publishes the operation
	provider types.ProviderType
	// Name identifies the specific operation within the provider
	name types.OperationName
```